source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -732,6 +732,54 @@ def _tag_by_mode(responses):
             _tag_journey_by_mode(j)
 
 
+def tag_robust_journeys(responses):
+    for r in responses:
+        for j in r.journeys:
+            if is_robust_journey(j):
+                j.tags.append("robust")
+
+
+robust_physical_modes = [
+    "physical_mode:RapidTransit",
+    "physical_mode:Metro",
+    "physical_mode:Train",
+    "physical_mode:RailShuttle",
+    "physical_mode:LocalTrain",
+    "physical_mode:LongDistanceTrain",
+]
+robust_fallback_modes = [response_pb2.Bike, response_pb2.Walking]
+# returns true if :
+#  - a journey has at least one public transport section
+#  - all public transport sections use robust physical modes
+#  - start and end fallbacks use robust fallback mode
+def is_robust_journey(journey):
+    found_a_robust_mode = False
+    found_a_non_robust_fallback = False
+    for section in journey.sections:
+
+        # check if this section uses a non robust fallback mode
+        if section.HasField("street_network"):
+            street_network = section.street_network
+            if street_network.HasField("mode"):
+                mode = street_network.mode
+                if mode not in robust_fallback_modes:
+                    found_a_non_robust_fallback = True
+                    break
+
+        if section.type != response_pb2.PUBLIC_TRANSPORT:
+            continue
+        if not section.HasField("uris"):
+            continue
+        uris = section.uris
+        if not uris.HasField("physical_mode"):
+            continue
+        if uris.physical_mode in robust_physical_modes:
+            found_a_robust_mode = True
+        else:
+            return False
+    return found_a_robust_mode and not found_a_non_robust_fallback
```

Comment:
It feels like all of this could be written with early return in mind: as soon as you find a non-robust section, `return False`. Then if all the sections have been checked for a non-robust condition, then you can `return True` (see also the above comment about returning early for fallback modes).

```suggestion
        if uris.physical_mode not in robust_physical_modes:
            return False
    return True
```

Reply:
As discussed, it's not enough since you also need to detect there is at least one section with a robust public transport... so you cannot do it without `found_a_robust_mode`. I'm resolving this comment.